sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/PCAExample.scala

```diff
@@ -15,59 +15,44 @@
  * limitations under the License.
  */
 
-// scalastyle:off println
 package com.intel.hibench.sparkbench.ml
 
-import org.apache.spark.SparkConf
-import org.apache.spark.SparkContext
-
-import org.apache.spark.mllib.feature.PCA
-import org.apache.spark.mllib.linalg.Vectors
-import org.apache.spark.mllib.regression.{LabeledPoint, LinearRegressionWithSGD}
-
+import org.apache.spark.ml.feature.{LabeledPoint, PCA}
 import org.apache.spark.rdd.RDD
 
-object PCAExample {
+import org.apache.spark.sql.SparkSession
 
+object PCAExample {
   def main(args: Array[String]): Unit = {
-    var inputPath = ""
-    var maxResultSize = "1g"
+    val spark = SparkSession
+      .builder
+      .appName("PCAExample")
+      .getOrCreate()
 
-    if (args.length == 2) {
+    var inputPath = ""
+    if (args.length == 1) {
       inputPath = args(0)
-      maxResultSize = args(1)
+      println(s"input Path: $inputPath")
+    } else {
+      System.err.println(
+        s"Usage: $PCADataGenerator <OUTPUT_PATH> <NUM_EXAMPLES> <NUM_FEATURES>"
+      )
+      System.exit(1)
     }
+    // Load training data
+    val data: RDD[LabeledPoint] = spark.sparkContext.objectFile(inputPath)
+    import spark.implicits._
+    val df = data.toDF()
 
-    val conf = new SparkConf()
-        .setAppName("PCAExample")
-        .set("spark.driver.maxResultSize", maxResultSize)
-    val sc = new SparkContext(conf)
-    
-    val data: RDD[LabeledPoint] = sc.objectFile(inputPath)
+    val pca = new PCA()
+      .setInputCol("features")
+      .setOutputCol("pcaFeatures")
+      .setK(3)
+      .fit(df)
 
-    val splits = data.randomSplit(Array(0.6, 0.4), seed = 11L)
-    val training = splits(0).cache()
-    val test = splits(1)
-
-    val pca = new PCA(training.first().features.size / 2).fit(data.map(_.features))
-    val training_pca = training.map(p => p.copy(features = pca.transform(p.features)))
-    val test_pca = test.map(p => p.copy(features = pca.transform(p.features)))
-
-    val numIterations = 100
-    val model = LinearRegressionWithSGD.train(training, numIterations)
-    val model_pca = LinearRegressionWithSGD.train(training_pca, numIterations)
-
-    val valuesAndPreds = test.map { point =>
-      val score = model.predict(point.features)
-      (score, point.label)
-    }
-
-    val valuesAndPreds_pca = test_pca.map { point =>
-      val score = model_pca.predict(point.features)
-      (score, point.label)
-    }
+    val result = pca.transform(df).select("pcaFeatures")
```

Comment:
This looks quite different to the original implementation. For example in the original code, it splits the data to train and test and do prediction. Why those was removed?

Reply:
Looks like we just need to benchmark PCA here, right?

Reply:
@carsonwang  yes , this is spark pca example , I didn't modified something ,but merely how to load training data. Therefore , I didn't split the data .